plugins/integrations/kubernetes-service/src/main/java/org/apache/cloudstack/api/command/user/kubernetes/cluster/StartKubernetesClusterCmd.java

```diff
@@ -99,6 +99,9 @@ public KubernetesCluster validateRequest() {
         if (kubernetesCluster == null) {
             throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Given Kubernetes cluster was not found");
         }
+        if (!kubernetesCluster.getManaged()) {
+            throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Given Kubernetes cluster is not managed by Apache CloudStack");
```

Comment:
any update on other CKS apis ?

[deleteKubernetesCluster (A)](https://cloudstack.apache.org/api/apidocs-4.18/apis/deleteKubernetesCluster.html)
[getKubernetesClusterConfig](https://cloudstack.apache.org/api/apidocs-4.18/apis/getKubernetesClusterConfig.html)
[listKubernetesClusters](https://cloudstack.apache.org/api/apidocs-4.18/apis/listKubernetesClusters.html)
[scaleKubernetesCluster (A)](https://cloudstack.apache.org/api/apidocs-4.18/apis/scaleKubernetesCluster.html)[startKubernetesCluster (A)](https://cloudstack.apache.org/api/apidocs-4.18/apis/startKubernetesCluster.html)
[startKubernetesCluster (A)](https://cloudstack.apache.org/api/apidocs-4.18/apis/startKubernetesCluster.html) (done)
[stopKubernetesCluster (A)](https://cloudstack.apache.org/api/apidocs-4.18/apis/stopKubernetesCluster.html)
[upgradeKubernetesCluster (A)](https://cloudstack.apache.org/api/apidocs-4.18/apis/upgradeKubernetesCluster.html)

Reply:
[deleteKubernetesCluster](https://cloudstack.apache.org/api/apidocs-4.18/apis/deleteKubernetesCluster.html) - Handled.
[getKubernetesClusterConfig](https://cloudstack.apache.org/api/apidocs-4.18/apis/getKubernetesClusterConfig.html) should return config for any cluster if the details are present.
[listKubernetesClusters](https://cloudstack.apache.org/api/apidocs-4.18/apis/listKubernetesClusters.html) - option to filter by managed.
[scaleKubernetesCluster](https://cloudstack.apache.org/api/apidocs-4.18/apis/scaleKubernetesCluster.html) -  validateKubernetesClusterScaleParameters fails if the cluster is not managed.
[startKubernetesCluster](https://cloudstack.apache.org/api/apidocs-4.18/apis/startKubernetesCluster.html) - Handled
[stopKubernetesCluster](https://cloudstack.apache.org/api/apidocs-4.18/apis/stopKubernetesCluster.html) - Handled
[upgradeKubernetesCluster](https://cloudstack.apache.org/api/apidocs-4.18/apis/upgradeKubernetesCluster.html) - validateKubernetesClusterUpgradeParameters fails if the cluster is not managed.

@weizhouapache I have verified the implementation, and I have handled all the APIs.

Reply:
great @vishesh92 
can you add `isManaged` to KubernetesClusterResponse, and show/hide actions on UI based on it ?

Reply:
@vishesh92 
also on UI, you can add a new field and a filter in the list view.

Reply:
Done